### ISSUE TITLE
Reorder Identifiers, Links, and Catalog Record boxes on record detail page

### ISSUE THREAD

**Issue by mkoo** [ArctosDB/arctos#5510]
Maybe considered part of[ #5509](https://github.com/ArctosDB/arctos/issues/5509)
(We've brought this up in the past but making sure we have an issue lodged.)

I have a quick sketch below but will write out the basic problem and solution. The problem is having the Identifiers listed in the bottom half of the GUID page, when we need to see this adjacent to the Agents. Instead, it makes more sense to have that listed next to the Links (external links like GBIF). Currently it looks like both Identifiers and Links would fit in a two-column row. The catalog record box should return to the bottom of the page as has data relevant to Arctos but not for the original data of the record.

[Link to sketch](https://docs.google.com/drawings/d/1lBUAVb_7V0VBK-yTT0v-vwIIGKuMhwCgR6b5mgeVY68/edit?usp=sharing)

![proposed record detail page change ](https://user-images.githubusercontent.com/2523089/213622891-3696b78f-9245-4789-8f74-3ad87cfa7a1d.jpg)

of course the real problem is that this is what we need for biological records and archival/cultural/ arts have other needs so in a future iteration we may need to  have two versions of this record page but right now it doesnt suit either.

tweak doable?

**Comment by campmlc** [comment]
I agree with this. We need to move the important data above the fold,
visible on most laptops. And collector and collector number should be
adjacent.

**Comment by AJLinn** [comment]
>  There are some position requests in there (@AJLinn and @ccicero that I can remember), 

My request is to keep media at top right with Agents next door.

> The catalog record box should return to the bottom of the page as has data relevant to Arctos but not for the original data of the record.

100% agree. At least for EH records, this should not be prioritized over the information about the cataloged item.

> Identifiers and parts have multiple possible views, and some need more room in certain directions than others.

Agree with @dustymc here as our part attribute contents are loving having room to breathe!

> It's definitely useful to find an example with Media.

For comparison, here's a [record with media and citations](https://arctos.database.museum/SpecimenDetailResponsive.cfm?collection_object_id=23809656) as well as multiple events and long content in the part attributes.

<img width="1259" alt="Screenshot 2023-01-20 at 8 42 58 AM" src="https://user-images.githubusercontent.com/17605945/213770871-8350b074-2c7e-4cea-b7d4-a74314c3d719.png">
<img width="1259" alt="Screenshot 2023-01-20 at 8 47 08 AM" src="https://user-images.githubusercontent.com/17605945/213770976-98f94367-2ac1-4987-b97f-93127f382d38.png">

Cultural collections, in my experience, rarely have links / relationships / other identifiers. We do have a lot in attributes and sometimes remarks. We're hoping to add more citations, but that's minimal for most of our records for now.

> We need to move the important data above the fold, visible on most laptops.

I'll continue to not hold my breath for that for cultural collections!

**Comment by campmlc** [comment]
Could citations be moved to the right of or under identifications, with
media below? Would it work for cultural collections to have agents moved
down adjacent to identifiers? Agents and identifiers need to be in
proximity. Or, put identifiers above Record Attributes, so there is a
chance of seeing them near the top? If cultural collections don't have
many, that won't take up much real estate?

**Comment by campmlc** [comment]
Or, looking back at @AJLinn s comment, Media could be top right opposite Identifications, followed by Citations, Agents, Identifiers, and Record Attributes, in whatever order fits best on the page? Does that still allow record attributes to be visible for the cultural collections?

**Comment by dustymc** [comment]
>right

Right and left don't really make sense in the context of this form. Given A B C, C is on the right (of the first row) if B exists and on the left (of the second row) if it doesn't, assuming two columns.

On a smaller screen things will just stack in a single column.

Adding a third (or more) column for even larger screens would be a relatively trivial media query (but it's about 20 times harder to organize than 2 columns so I didn't spend too much time on the idea).

>citations be moved to the right of ... identifications,

Identifications is currently full-span, so there is no "right of." Under is available.

>agents moved down adjacent to identifiers

Identifiers are also full span, "before" and "after" are the plausible relationships - and "after" will result in a tiny thing and then a whole bunch of nothing, unless you're putting more maybe-there partial-span stuff down too.

>opposite Identifications

Before (which would result in most of the prime real estate being mostly empty for most records, unless other stuff which might fill it is included) and after are the only available relationships to identifications.

>in whatever order fits best on the page

If you're asking for the elements to somehow sort themselves, I think that would be magic.

If you mean "whatever we might agree to" then sure, but be aware that that's 2 full-span and 2 partial-span elements, so ordering can result in half-screen gaps, and so can missing elements.

I think what I need is a list of the elements in order without left/right/above/below references, but you need to be aware of the size (only two possibilities at the moment) of the various things while assembling the list.

Is the UI Task Force done? Maybe it should be resurrected for this.

**Comment by campmlc** [comment]
If we have the time, I absolutely agree with another UI meeting. @AJLinn can you make a meeting next Tuesday at 4 pm MDT?

**Comment by mkoo** [comment]
I'll add a 3rd UI meeting for this Tuesday on the Arctos calendar. Let us know if you cannot make it and please suggest a better time for you @AJLinn

**Comment by AJLinn** [comment]
I just need to arrange for my kid to get picked up from school, unless you all can do an hour earlier?

**Comment by mkoo** [comment]
I can-- Teresa wont be able to make it. Let's shift for this week since it's priority work. We will encapsulate meeting in this thread and still be able to have discussion esp. if anything radical comes up but it seems we're in the finesse stage now

Calendar will be updted accordingly!

**Comment by AJLinn** [comment]
thanks for accommodating me this time! See you Tuesday!

**Comment by Jegelewicz** [comment]
> And collector and collector number should be adjacent.

https://github.com/ArctosDB/arctos/issues/4604

**Comment by dustymc** [comment]
Agree with @Jegelewicz - if there's really some link then lets form a task force and figure out how to actually do that as part of @4604, if there's not then the symbolic closeness (or whatever it is...) shouldn't be necessary.

**Comment by dustymc** [comment]
* can IDs/relationships not exist when no data?
* Provide list of elements and column requirements
* links - all the way to the bottom
*

**Comment by dustymc** [comment]
HEADER

make sure inst/collection are getting link css
see angelo - image has weird padding, figure out why and make it die

**Comment by campmlc** [comment]
I'm ok with this:

   - can IDs/relationships not exist when no data?

**Comment by dustymc** [comment]
Here are elements and layout:

Identifications: full width
Agents (including verbatim) - half-row
Citations  - half-row
Links (going to bottom)
TAGs - half-row
Media - half-row
Catalog Record/Curatorial - half-row
Record Attributes - full row
Place- full, includes locality and event attributes when present
Identifiers and Relationships - full, user-customizable
Parts - full, user-customizable
Reports - full

**Comment by campmlc** [comment]
How do we link identifiers, relationships, and agents for collector number,
preparation number?

**Comment by mkoo** [comment]
> How do we link identifiers, relationships, and agents for collector number, preparation number?

@campmlc please file a new issue so we dont lose track of your question if unresolved!

**Comment by mkoo** [comment]
> Here are elements and layout:

>Identifications: full width
>Agents (including verbatim) - half-row
Citations - half-row
Links (going to bottom)
TAGs - half-row
Media - half-row
Catalog Record/Curatorial - half-row
Record Attributes - full row
Place- full, includes locality and event attributes when present
Identifiers and Relationships - full, user-customizable
Parts - full, user-customizable
Reports - full

Thanks for this list-- useful-- please let us know when you reorder priority of display of GUID prior to launch tomorrow night! Thanks! 

proposed ordered list:

1.  Identifications: full width
2. Agents (including verbatim) - half-row
3. Citations - half-row
4. Identifiers and Relationships - full, user-customizable
5. TAGs - half-row
6. Media - half-row
7. Catalog Record/Curatorial - half-row (should this be full row since it is not paired with another half-row? Esp. if cultural collections use this for a lot of remark details that are public
8. Record Attributes - full row
9. Place- full, includes locality and event attributes when present
10. Parts - full, user-customizable
11. Reports - full
12. Links (going to bottom)

**Comment by dustymc** [comment]
I did this in the hopes that it will help understand how this works. Mixing up the single-column things in with the 2-column things just inevitably results in a lot of empty whitespace. Pick a number, search for it in catalog number, open a bunch of tabs, almost all of them will look something like...

<img width="1579" alt="Screenshot 2023-01-25 at 5 43 31 PM" src="https://user-images.githubusercontent.com/5720791/214737703-2ecea919-271d-417a-8e2a-12faf384205c.png">

and I've got to scroll past acres of empty nothingness to find any information. 

No bueno.

Let me know when I can move all the single-column things back to the top of the page.

**Comment by dustymc** [comment]
For the sake of fairness: https://arctos-test.tacc.utexas.edu/guid/MVZ:Egg:6 looks pretty good with this layout.

**Comment by mkoo** [comment]
remind me again why Identifiers is NOT a half-row ?
(would save a half-acre of fallow web page...)

Yea and that Egg record is typical of a lot bio records so exactly what we were talking about Agents and Identifiers nice and convenient to each other. thanks for being fair!

**Comment by ewommack** [comment]
> remind me again why Identifiers is NOT a half-row ?

Is that because it fills out huge with some of the collections that have a lot of info in there (cultural)?

**Comment by jldunnum** [comment]
Agreed that layout looks good!

______________________________________________________________

Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
<img width="1285" alt="Screenshot 2023-01-25 at 6 49 59 PM" src="https://user-images.githubusercontent.com/5720791/214747671-8a755de0-9d43-409b-804d-ba4becc9b3e4.png">

Even simple mostly-legacy records with good IDs don't fit in a half-page. Parasites are worse, the coming metadata will be much worse. 

>Egg record is typical

Citations and Media and not much else are not typical!

Typical is agents (and no other single-column things), then nothing where identifiers breaks the page up, the one more 2-column thing.

**Comment by campmlc** [comment]
Absolutely agree with @dustymc , for parasites and pathogens and genetic and genomic data, identifiers need to be a full row, as we need the repeatedly requested additional metadata to include remarks and ID issued by (e.g., for MVZ, you'd have the institutional agent name). Maybe that seems obvious for MVZ, but for instituions such as ZRC it is not. 
If we had the ID Issued by  metadata, then for collector number and preparator number, the Issued by agent would be the collector and preparator's agent name, and all the data would be together in one place.
We could also specify that an ID was issued by NCBI, and capture all the various types of GenBank identifiers, instead of only a single one right now which doesn't accommodate their data complexity.
And we could simplify the identifiers table. Every external institution would not need an identifier type, as the institution would be the issued by agent. And from the institution's agent summary page, we could find all the records that agent was responsible for an "Issued by" determination.

**Comment by mkoo** [comment]
Actually I'm ok with Identifiers for full row too if everyone can live with the potential for white space-- it actually doesnt bother me that much as I check out records all last night (not just MVZ's but ALL our collection types- fun!). There's still info on every row and I just hit my jump to bottom shortcuts if  I dont want to scroll.

I suggest we keep the current order b/c it's still important to have Identifiers prioritized near the top.
